### PR TITLE
[Fix][Connector-V2][OceanBase] Remove OceanBase catalog's dependency on mysql driver

### LINKED ISSUE #7289: [Improve][Connector-V2][OceanBase] Remove OceanBase catalog's dependency on mysql driver
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Now the OceanBase catalog directly extends the implementation of mysql catalog and relies on the data type class provided by mysql driver.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xxsc0529**
### Purpose of this pull request

The original oceanbase catalog directly inherits the mysql catalog, which results in the need to add an additional mysql driver to the lib directory when using oceanbase alone. This submission solves this problem.

### Does this PR introduce _any_ user-facing change?

yes，Close #7289

### How was this patch tested?

Here is a simple test example
env {
  job.mode = "BATCH"
}

source {
  jdbc {
    url = "jdbc:oceanbase://localhost:2881/demo"
    driver = "com.oceanbase.jdbc.Driver"
    user = "xxxx@tt1"
    password = "123456"
    generate_sink_sql = true
    compatible_mode="mysql"
    database = "demo"
    table = "t_ds_user"
    query = "select * from t_ds_user"
  }
}

transform {
}

sink {
  jdbc {
    url = "jdbc:oceanbase://localhost:2881/test"
    driver = "com.oceanbase.jdbc.Driver"
    user = "xxxx@tt1"
    password = "123456"
    compatible_mode="mysql"
    generate_sink_sql = true
    database = "test"
    table = "t_ds_user"
  }
Before the code was modified, because the oceanbase catalog inherited the mysql catalog, some classes could not be found when the mysql dependency package was not introduced, such as
image
It is also impossible to automatically create a table
After making the modification, the table can be created successfully without error.
image
image

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by xxsc0529** [comment]
> ### Purpose of this pull request
> The original oceanbase catalog directly inherits the mysql catalog, which results in the need to add an additional mysql driver to the lib directory when using oceanbase alone. This submission solves this problem.
> 
> ### Does this PR introduce _any_ user-facing change?
> yes，colse #7289
> 
> ### How was this patch tested?
> Here is a simple test example env { job.mode = "BATCH" }
> 
> # 配置数据源
> source { jdbc { url = "jdbc:oceanbase://localhost:2881/demo" driver = "com.oceanbase.jdbc.Driver" user = "xxxx@tt1" password = "123456" generate_sink_sql = true compatible_mode="mysql" database = "demo" table = "t_ds_user" query = "select * from t_ds_user" } }
> 
> transform { }
> 
> # 配置目标库
> sink { jdbc { url = "jdbc:oceanbase://localhost:2881/test" driver = "com.oceanbase.jdbc.Driver" user = "xxxx@tt1" password = "123456" compatible_mode="mysql" generate_sink_sql = true database = "test" table = "t_ds_user" } Before the code was modified, because the oceanbase catalog inherited the mysql catalog, some classes could not be found when the mysql dependency package was not introduced, such as image It is also impossible to automatically create a table After making the modification, the table can be created successfully without error. image image
> 
> ### Check list
> * [ ]  If any new Jar binary package adding in your PR, please add License Notice according
>   [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
> * [ ]  If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
> * [ ]  If you are contributing the connector code, please check that the following files are updated:
>   
>   1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
>   2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
>   3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
> * [ ]  Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

**Comment by Hisoka-X** [comment]
cc @whhe

**Comment by Hisoka-X** [comment]
Hi @xxsc0529 , does oceanbase has own jdbc driver?

**Comment by xxsc0529** [comment]
> Hi @xxsc0529 , does oceanbase has own jdbc driver?

yes You need to use the oceanbase-client package  
    <dependency>
        <groupId>com.oceanbase</groupId>
        <artifactId>oceanbase-client</artifactId>
        <version>2.4.3</version>
    </dependency>
https://github.com/oceanbase/obconnector-j.git
It is recommended that you use the oceanbase-client driver to connect to OceanBase

**Comment by Hisoka-X** [comment]
Please follow the guide to open CI on your fork repository.
image

**Comment by xxsc0529** [comment]
> Please follow the guide to open CI on your fork repository. image

done

**Comment by Hisoka-X** [comment]
Please push an emtpy commit to trigger ci

**Comment by Hisoka-X** [comment]
What's do you think about this PR? @hailin0

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java [hunk 1]
```diff
@@ -17,23 +17,191 @@
...
+
+@Slf4j
+public class OceanBaseMySqlCatalog extends AbstractJdbcCatalog {
+
+    private static final String SELECT_COLUMNS_SQL_TEMPLATE =
+            "SELECT * FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_SCHEMA = '%s' AND TABLE_NAME ='%s' ORDER BY ORDINAL_POSITION ASC";
 
     static {
         SYS_DATABASES.clear();
-        SYS_DATABASES.add("information_schema");
```
Comment (whhe) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: Why delete them?

**Review by whhe — COMMENTED** [review]
I just noticed that in JdbcDialect there is a method `decorateWithComment` that references MysqlType. I think it should also be removed. WDYT @Hisoka-X @xxsc0529

**Comment by xxsc0529** [comment]
> decorateWithComment

OceanBase has rewritten a method called decorateOceanBaseWithComment，Removed MysqlType

**Comment by xxsc0529** [comment]
> 我刚刚注意到 JdbcDialect 中有一个`decorateWithComment`引用 MysqlType 的方法。我认为它也应该被删除。WDYT@Hisoka-X @xxsc0529

OceanBase has rewritten a method called decorateOceanBaseWithComment，Removed MysqlType

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java (hunk 1 above)
Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: done

Comment (whhe) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: It's better to rollback the code block to reduce redundant git changes.

**Comment by whhe** [comment]
If you agree with me, I think we can create a new PR to remove all MysqlType references in JdbcDialect. @Hisoka-X @xxsc0529

**Comment by Hisoka-X** [comment]
> If you agree with me, I think we can create a new PR to remove all MysqlType references in JdbcDialect. @Hisoka-X @xxsc0529

+1. cc @hailin0

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java [hunk 2]
```diff
@@ -666,6 +667,16 @@ default String decorateWithComment(
...
 
+    default String decorateOceanBaseWithComment(
+            String basicSql, BasicTypeDefine<OceanBaseMysqlType> typeBasicTypeDefine) {
+        String comment = typeBasicTypeDefine.getComment();
+        StringBuilder sql = new StringBuilder(basicSql);
+        if (StringUtils.isNotBlank(comment)) {
+            sql.append("COMMENT '").append(comment).append("'");
+        }
+        return sql.toString();
+    }
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: It is strange to add method for oceanbase in jdbcdiaect. I think we should `remove all MysqlType references in JdbcDialect`.  https://github.com/apache/seatunnel/pull/7311#issuecomment-2272571440

Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/JdbcDialect.java]: Modified according to the latest code

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java (hunk 1 above)
Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/oceanbase/OceanBaseMySqlCatalog.java]: ok

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMysqlType.java [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.jdbc.internal.dialect.oceanbase;
+
+import com.mysql.cj.util.StringUtils;
```
Comment (whhe) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMysqlType.java]: Have you tested creating table without mysql driver? IMO this util shoud not be introduced here.

Comment (xxsc0529) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/oceanbase/OceanBaseMysqlType.java]: I'm sorry for the error caused by the code rollback, I've modified it and retested it，There is no MySQL driver in this lib, only Oceanbase, which can automatically create tables
image
image

**Review by whhe — APPROVED** [review]

**Review by Hisoka-X — COMMENTED** [review]
Please remove mysql driver in https://github.com/apache/seatunnel/blob/a18fca80061925ba3a2c9bd152b450dcf3d9308b/seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMysqlIT.java#L75. And add oceanbase driver to make sure it work fine.

**Comment by xxsc0529** [comment]
> JdbcOceanBaseMysqlIT.java

done

**Review by Hisoka-X — DISMISSED** [review]
Thanks @xxsc0529 !. LGTM if ci passes. cc @hailin0

**Comment by Hisoka-X** [comment]
Thanks @whhe for review!

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMysqlIT.java [hunk 4]
```diff
@@ -68,12 +68,12 @@ public class JdbcOceanBaseMysqlIT extends JdbcOceanBaseITBase {
                                 "mkdir -p /tmp/seatunnel/plugins/Jdbc/lib && cd /tmp/seatunnel/plugins/Jdbc/lib && wget "
                                         + driverUrl()
                                         + " && wget "
-                                        + mysqlDriverUrl());
+                                        + oceanbaseDriverUrl());
                 Assertions.assertEquals(0, extraCommands.getExitCode(), extraCommands.getStderr());
             };
 
-    String mysqlDriverUrl() {
-        return "https://repo1.maven.org/maven2/com/mysql/mysql-connector-j/8.0.32/mysql-connector-j-8.0.32.jar";
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMysqlIT.java:76]: Is it compatible with older versions?

Comment (xxsc0529) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/connector-jdbc-e2e-part-2/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcOceanBaseMysqlIT.java:76]: Yes, you can use the old version or the new version, and only need oceanbase-client

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @xxsc0529 !

**Review by hailin0 — APPROVED** [review]

**Pull request merged**